jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -1386,7 +1386,11 @@ public String getServletPath()
         // INCLUDE dispatch, in which case this method returns the servletPath of the source servlet,
         // which we recover from the IncludeAttributes wrapper.
         ServletPathMapping mapping = findServletPathMapping();
-        return mapping == null ? "" : mapping.getServletPath();
+        // per Servlet spec 3.5, mapping servlet paths of "/*", "/", and "" should return ""
+        if ((mapping == null) || mapping.getServletPath() == null || (mapping.getServletPath().equals("")) || mapping.getServletPath().equals("/") || (mapping.getServletPath().equals("/*")))
+            return "";
+        else
+            return mapping.getServletPath();
```

Comment:
I don't think this is the correct fix.  The servletPath returned from the mapping should be "" for a root match.  We can see the code [intends to make it so](https://github.com/eclipse/jetty.project/blob/jetty-10.0.5/jetty-server/src/main/java/org/eclipse/jetty/server/ServletPathMapping.java#L51-L56).  

I'm seeing the mapping for that as:
```
ServletPathMapping{matchValue=, pattern=, servletName=org.eclipse.jetty.servlet.ServletContextHandlerTest$3-2bdd8394, mappingMatch=null, servletPath=null, pathInfo=null}
``` 
So the question is, why is this not seen as a ROOT match?  Why is the mappingMatch null?

Reply:
Ah, as the OP said, it looks to be an issue with the pathInContext being null when constructing the ServletPathMapping???

Reply:
Oh! We are treating EXACT and ROOT matches the same.... but ROOT matches have no prefix. 
So I can see a fix... but not sure I like it..... standby I'll update branch if it passed tests....

Reply:
@joakime  I've pushed an alternate fix that corrects the ROOT match so the mapping servletPath is also correct.
It works... but something feels off about it still???  I don't like forcing the path.  Maybe the prefix of a root pathspec should be "" ?